db.go

```diff
@@ -265,7 +273,12 @@ func readUnderAccounts(tree *avl.Tree, id AccountID, key []byte) ([]byte, bool)
 }
 
 func writeUnderAccounts(tree *avl.Tree, id AccountID, key, value []byte) {
-	tree.Insert(append(keyAccounts[:], append(key, id[:]...)...), value[:])
+	k := make([]byte, 0, len(keyAccounts)+len(key)+len(id))
+	k = append(k, keyAccounts[:]...)
+	k = append(k, key...)
+	k = append(k, id[:]...)
+
```

Comment:
How does it compare to this?

```go
func main() {
	a := []byte("hello")
	b := []byte(" there")
	c := []byte(" how are you?")
	fmt.Println(append(a, append(b, c...)...))
}
```

Reply:
(wondering if there are compiler optimizations for this)

Reply:
The code: https://gist.github.com/AhmadMuzakkir/71d0219e8d6183525fd1e93959c7d8a0

I ran the benchmarks 100 times.

Results:
```
name           old time/op    new time/op    delta
SliceAppend-8     568ns ± 7%     542ns ±10%   -4.52%  (p=0.000 n=100+100)

name           old alloc/op   new alloc/op   delta
SliceAppend-8     96.0B ± 0%     64.0B ± 0%  -33.33%  (p=0.000 n=100+100)

name           old allocs/op  new allocs/op  delta
SliceAppend-8      3.00 ± 0%      2.00 ± 0%  -33.33%  (p=0.000 n=100+100)
```